Jenkinsfile.cloud

```diff
@@ -55,20 +81,32 @@ node(env.NODE) {
         """
     }
 
-    def commit, dirpath, qcow
+    def commit, version, dirpath, qcow, vmdk
     stage("Finalizing") {
         def image = sh(returnStdout: true, script: "ls /var/lib/imagefactory/storage/*.body").trim()
         // just introspect after the fact to avoid race conditions
         commit = sh(returnStdout: true,
                     script: """LIBGUESTFS_BACKEND=direct virt-cat -a ${image} -m /dev/coreos/root:/ \
                                     /ostree/repo/refs/remotes/rhcos/${ref}""").trim()
-        currentBuild.description = 'commit ' + commit
+
+        // do this again *after* running imgfac to close race
+        // fetch parent too for the later db diff
+        sh "ostree --repo=repo pull --depth=1 --commit-metadata-only tmp ${commit}"
+        version = sh(returnStdout: true,
+                     script: "ostree show --repo=repo --print-metadata-key=version ${commit}").trim()
+        assert (version.startsWith("'") && version.endsWith("'"))
+        version = version[1..-2] // trim single quotes
+
+        currentBuild.description = "${version} (${commit})"
 
         def dirname = (new Date()).format("YYYY-MM-dd-HH-mm-ss-") + commit
         dirpath = "${output}/cloud/${dirname}"
         qcow = "${dirpath}/rhcos.qcow2"
         vmdk = "${dirpath}/rhcos.vmdk"
         sh "mkdir -p ${dirpath}"
+        // this belongs better in a JSON file, but for now just use a file;
+        // this is used higher up to determine no-op changes
+        sh "echo '${version}' > ${dirpath}/version.txt"
```

Comment:
I didn't want to make the PR even longer, but in a future patch we can just output a mini JSON file with all the metadata in there.